### PR TITLE
Patch so that "make install" works properly.

### PR DESCRIPTION
**Pull request by rhoneyager-tomorrow**
## Description

The ecbuild to cmake update broke the ability to have CMake "install" CRTM. This fixes it. The patch is based on the [v3.1.0-skylabv8](https://github.com/JCSDA/CRTMv3/releases/tag/v3.1.0-skylabv8) tag, which is why there are conflicts against `develop`.

Expected behavior:
```bash
cmake -D CMAKE_INSTALL_PREFIX=/some/installation/directory   path_to_crtm_source
make install

-- Install configuration: "Release"
-- Up-to-date: /some/installation/directory/lib64/libcrtm.a
-- Up-to-date: /some/installation/directory/lib64/cmake/crtm/crtm-config.cmake
-- Up-to-date: /some/installation/directory/lib64/cmake/crtm/crtm-config-version.cmake
-- Up-to-date: /some/installation/directory/lib64/cmake/crtm/crtm-targets.cmake
-- Up-to-date: /some/installation/directory/lib64/cmake/crtm/crtm-targets-release.cmake
-- Up-to-date: /some/installation/directory/module/crtm/Intel/2021.9.0.20230302
-- Up-to-date: /some/installation/directory/module/crtm/Intel/2021.9.0.20230302/crtm_module.mod
......
```

## Issue(s) addressed

None tracked here.

## Dependencies

None.

## Impact

Spack can build CRTM v3 again.

## Checklist

- [x] I have performed a self-review of my own code
- [x] ~I have made corresponding changes to the documentation~ (No doc changes relevant)
- [x] ~I have run the unit tests before creating the PR~ (None applicable)
- [x] I have tested this code in a jedi-bundle build
- [x] I have built CRTM independent of a bundle, run `make install`, and then passed that installation directory for use in a bundle build of ufo. Ex: `cmake -Dcrtm_DIR=/some/installation/directory/for/crtm`.
- [x] I have built CRTM independent of a bundle, and passed the CRTM build directory for use in a bundle build of ufo. Ex: `cmake -Dcrtm_DIR=/some/build/directory/for/crtm`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -70,14 +70,9 @@ install(FILES
 # Install the export set for use with the install-tree
 install(EXPORT ${PROJECT_NAME}-config
 	FILE "${PROJECT_NAME}-targets.cmake"
-	#FILE "${PROJECT_NAME}-config.cmake"
```
Comment (BenjaminTJohnson) [inline CMakeLists.txt]: as far as I understand it, the crtm-config.cmake generation is necessary for integration in JEDI/UFO -- that's what it looks for to find CRTM.   I may be missing something.

**Comment by BenjaminTJohnson** [comment]
@rhoneyager-tomorrow just tested and looks like it works without any issues.

**Review by BenjaminTJohnson — APPROVED** [review]

**Comment by rhoneyager-tomorrow** [comment]
Still working on it! It's in draft and visible so that Rick Grubin can also take a look and finish any leftover parts.

**Comment by rhoneyager-tomorrow** [comment]
Open for review.

You have to consider three cases for how CMake can be built and linked with dependent projects.
1. A jedi-bundle build. This is managed by ecbuild, and a whole slew of different repositories are combined together as a set of CMake projects and sub-projects. There's nothing special in this case.
2. Linking a downstream project to a CMake **build** directory. At your build root, the `crtm-config.cmake` and `crtm-targets*.cmake` files are written, and `find_package(crtm)` will locate crtm if you give it a hint (by setting `crtm_DIR` to the crtm build location).
3. Linking a downstream project to a CMake **install** directory. This is the collection of files and folders written when you run `make install`. Normally, this defaults to `/usr/local`, but this install path may be customized by setting `CMAKE_PREFIX_PATH` accordingly. As with case 2, `find_package(crtm)` will locate crtm if you give it a hint (by setting `crtm_DIR` to the crtm install location). This is also how Spack installs packages. It sets the correct CMake variables and runs `make install`, and everything just works.

I've tested all three cases, and can successfully link crtm to ufo in each case.

**Review by BenjaminTJohnson — APPROVED** [review]

**Pull request merged**